internal/controllers/requirements/requirements.go

```diff
@@ -27,8 +27,8 @@ import (
 var ErrRequirements = errors.New("check the requirements for run and try again")
 
 type Requirements struct {
-	gitRequirements    *git.RequirementGit
-	dockerRequirements *docker.RequirementDocker
+	gitRequirements    git.IRequirementGit
+	dockerRequirements docker.IRequirementDocker
```

Comment:
Interface should be declared on the side who depends, instead who implements. 

So `IRequirementGit` and `IRequirementDocker` should be declared on `internal/controllers/requirements` package.

[Check this](https://github.com/golang/go/wiki/CodeReviewComments#interfaces)

Reply:
I think that we even don't need this interface, since the `RequirementsGit` and `RequirementDocker` docker don't have any field, we can declare a function type that both git and docker implements. For example:

```go
//  On internal/controllers/requirements/requirements.go

type ValidationFn func() error

type Requirements struct {
	gitValidiation    ValidationFn
	dockerValidation ValidationFn
}

func (r *Requirements) ValidateDocker() error {
	return r.dockerValidation()
}

func (r *Requirements) ValidateGit() error {
	return r.gitValidiation()
}

```

On tests we can pass a mock function to test error results. 

What do you think about this approach?

Reply:
this is a great idea, i'll do that